contracts/PBAB+Collabs/GenArt721CoreV2_PBAB_FLEX.sol

```diff
@@ -0,0 +1,893 @@
+    {
+        projects[_projectId].scripts[
+            projects[_projectId].scriptCount
+        ] = _script;
+        projects[_projectId].scriptCount = projects[_projectId].scriptCount + 1;
+    }
+
+    /**
+     * @notice Updates script for project `_projectId` at script ID `_scriptId`.
+     * @param _projectId Project to be updated.
+     * @param _scriptId Script ID to be updated.
+     * @param _script Script to be added.
+     */
+    function updateProjectScript(
+        uint256 _projectId,
+        uint256 _scriptId,
+        string memory _script
+    ) public onlyUnlocked(_projectId) onlyArtistOrWhitelisted(_projectId) {
+        require(
+            _scriptId < projects[_projectId].scriptCount,
+            "scriptId out of range"
+        );
+        projects[_projectId].scripts[_scriptId] = _script;
+    }
+
+    /**
+     * @notice Updates external asset dependency for project `_projectId`.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index.
+     * @param _cid Asset cid (Content identifier).
+     * @param _dependencyType Asset dependency type.
+     *  0 - IPFS
+     *  1 - ARWEAVE
+     */
+    function updateProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index,
+        string calldata _cid,
+        ExternalAssetDependencyType _dependencyType
+    )
+        external
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projects[_projectId].externalAssetDependencyCount,
+            "Asset index out of range"
+        );
+        projects[_projectId].externalAssetDependencies[_index].cid = _cid;
+        projects[_projectId]
+            .externalAssetDependencies[_index]
+            .dependencyType = _dependencyType;
+        emit ExternalAssetDependencyUpdated(
+            _projectId,
+            _index,
+            _cid,
+            _dependencyType
+        );
+    }
+
+    /**
+     * @notice Removes external asset dependency for project `_projectId` at index `_index`.
+     * Assets with indices higher than `_index` can have their indices adjusted as a result of this operation.
+     * @param _projectId Project to be updated.
+     * @param _index Asset index
+     */
+    function removeProjectExternalAssetDependency(
+        uint256 _projectId,
+        uint256 _index
+    )
+        external
+        onlyUnlockedProjectExternalAssetDependencies(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        require(
+            _index < projects[_projectId].externalAssetDependencyCount,
+            "Asset index out of range"
+        );
+
+        uint256 lastElementIndex = projects[_projectId]
+            .externalAssetDependencyCount - 1;
+
+        projects[_projectId].externalAssetDependencies[_index] = projects[
+            _projectId
+        ].externalAssetDependencies[lastElementIndex];
```

Comment:
I'm pretty sure this is okay and will deep-copy the entire struct at index `lastElementIndex` to the struct at index `_index`.
@lyaunzbe - would appreciate if you double checked this ref and make sure you agree: https://docs.soliditylang.org/en/develop/types.html#data-location-and-assignment-behaviour

Also would be good to double check that tests ensure this deepcopys as expected.

Reply:
@lyaunzbe could you try to take a look at this one today and then I'll do a final review pass over this PR as a sanity check and think we can 🚢 merging this.

Reply:
https://github.com/ArtBlocks/artblocks-contracts/blob/9d09585301e214dc8d54d521ad4a610b987bb0b7/test/core/GenArt721CoreV2_PBAB_FLEX_Integration.tests.ts#L134 tested here that deepcopy as working as expected 👍